UE/Tests/Application/Ports/UserPortTestSuite.cpp

```diff
@@ -54,7 +57,33 @@ TEST_F(UserPortTestSuite, shallShowMenuOnConnected)
     EXPECT_CALL(guiMock, setListViewMode()).WillOnce(ReturnRef(listViewModeMock));
     EXPECT_CALL(listViewModeMock, clearSelectionList());
     EXPECT_CALL(listViewModeMock, addSelectionListItem(_, _)).Times(AtLeast(1));
+    EXPECT_CALL(guiMock, setAcceptCallback(_));
     objectUnderTest.showConnected();
 }
 
+TEST_F(UserPortTestSuite, shallViewSmsList)
+{
+    using SmsMessages = std::vector<std::pair<Sms, SmsState>>;
+    const SmsMessages smsMessages;
+
+    EXPECT_CALL(guiMock, setListViewMode()).WillOnce(ReturnRef(listViewModeMock));
+    EXPECT_CALL(listViewModeMock, clearSelectionList());
+    EXPECT_CALL(listViewModeMock, addSelectionListItem(_, _)).Times(AtLeast(0));
+    EXPECT_CALL(guiMock, setAcceptCallback(_));
+    EXPECT_CALL(guiMock, setRejectCallback(_));
+    objectUnderTest.viewSmsList(smsMessages);
+}
+
+TEST_F(UserPortTestSuite, shallViewSms)
+{
+    const Sms sms{common::PhoneNumber{112},"example"};
+
+    EXPECT_CALL(guiMock, setViewTextMode()).WillOnce(ReturnRef(textModeMock));
+    EXPECT_CALL(textModeMock, setText(_));
```

Comment:
put text of sms to setText method

Reply:
i think, unexpected call error might appear